### PR TITLE
DOCSP-43664: v1.17 release (other changes)

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-43664>
Staging - https://deploy-preview-384--docs-golang.netlify.app/compatibility/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/includes/mongodb-compatibility-table-go.rst [hunk 1]
```diff
@@ -14,24 +14,6 @@
      - MongoDB 4.0
      - MongoDB 3.6
      - MongoDB 3.4
```
Comment (mongoKart) [inline source/includes/mongodb-compatibility-table-go.rst:16]: S: It looks like v3.4 can also be removed

**Review by mongoKart — APPROVED** [review]
LGTM w/ a suggestion

File: source/whats-new.txt [hunk 2]
```diff
@@ -171,6 +172,37 @@ This release includes the following improvements and fixes:
...
 
+.. _golang-version-1.17:
+
+What's New in 1.17
+------------------
+
+.. important::
+
+   v1.17 is the final planned v1.x version release. This version will
+   receive bug fixes, but future development and features will be
```
Comment (prestonvasquez) [inline source/whats-new.txt:183]: We will also fix security issues such as CVEs which are different from bugs. Suggest updating the language here: 

> receive security and bug fixes, but future development and features will be

File: source/whats-new.txt [hunk 3]
```diff
@@ -171,6 +172,37 @@ This release includes the following improvements and fixes:
...
+   included only in 2.x versions of the driver.
+
+The 1.17 {+driver-short+} release includes the following improvements
+and fixes:
```
Comment (prestonvasquez) [inline source/whats-new.txt:187]: Should we specifically mention that this minor release includes partial support for MongoDB 8.0 (with the exception of bulk write)?

**Review by prestonvasquez — CHANGES_REQUESTED** [review]

File: source/whats-new.txt (hunk 3 above)
Comment (rustagir) [reply source/whats-new.txt:187]: Yes, I can add here and in the compat table

**Review by prestonvasquez — APPROVED** [review]

**Pull request merged**